### PR TITLE
Support Opset 20 for ReduceMin and ReduceMax

### PR DESCRIPTION
**Pull request by hamptonm1**
ReduceMin and ReduceMax now support data type `tensor(boolean)` for opset20. I updated the files accordingly, so that we can get things updated. Thanks! 

We need to have legacy support for both versions 18 and 13 to bypass the errors I was receiving in the backend tests because the backend tests for ReduceMin and ReduceMax are not separated by versions. So I had to update the decompose pass for version 18 to version 20.

### CODE REVIEW

**Comment by hamptonm1** [comment]
@chentong319 @AlexandreEichenberger Let me know what you guys think and if I should make any further updates. Thanks!

File: src/Dialect/ONNX/ONNXOps/Math/Reduction.cpp
Comment (philass) [inline src/Dialect/ONNX/ONNXOps/Math/Reduction.cpp:278]: Shouldn't this be 
```
return inferShapeForReductionOps<ONNXReduceMaxV18Op>(*this);
```

File: src/Dialect/ONNX/ONNXOps/Math/Reduction.cpp
Comment (philass) [inline src/Dialect/ONNX/ONNXOps/Math/Reduction.cpp:323]: Same here

**Review by philass — APPROVED** [review]
Other than that LGTM

It's a shame that a new op needs to be added for this, but I see how our current infrastructure makes that a necessity. This is beyond the scope of this PR, but we should rethink our versioning. In cases where the newer version is a super set of functionality with no breaking changes, we shouldn't have to add new ops.

**Review by AlexandreEichenberger — CHANGES_REQUESTED** [review]
Code seems fine but I would like to know why a new op is needed before we merge, as we should avoid new versions as much as possible

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM after explanation, it makes sense.

**Comment by hamptonm1** [comment]
> LGTM after explanation, it makes sense.

@AlexandreEichenberger Thanks for the talk and I am glad everything makes sense now. Side point for reference if anyone wants to understand the reason why please view the description in this PR for clarity. Thanks!!

File: src/Dialect/ONNX/ONNXOps/Math/Reduction.cpp
Comment (hamptonm1) [reply src/Dialect/ONNX/ONNXOps/Math/Reduction.cpp:278]: @philass Thanks for catching that!

**Pull request merged**
